src/components/fileSelect/FileSelect.tsx

```diff
@@ -0,0 +1,276 @@
+            if (files && files.length > 0) {
+                for (const key in files) {
+                    // Condition check to ignore the Prototypes in the files object
+                    if (files.hasOwnProperty(key)) {
+                        fileNames.push(files[key].name);
+                        fileObjects.push(files[key]);
+                    }
+                }
+                this.setState(
+                    {
+                        fileName: fileNames.join(","),
+                        files: fileObjects,
+                    },
+                    () => {
+                        this.props.onFileChange(this.state.files);
+                    },
+                );
+            }
+        }
+    };
+    // Function which triggers edit modal when the edit icon is clicked
+    editFiles = () => {
+        this.setState({
+            isEditMode: true,
+        });
+    };
+
+    // Function triggered when file name is changed in any of the file's text box
+    onFileNameChange = (evt: ChangeEvent<HTMLInputElement>, index: number) => {
+        if (evt.target) {
+            const editedValue = evt.target.value;
+            const {files} = this.state;
+            // When new file name is same as existing file name, disabling the rename button.
+            if (editedValue === files[index].name) {
+                this.setState({
+                    editedFileName: "",
+                    enableButton: ENABLE_BUTTON_DEFAULT,
+                });
+            }
+            // Enabling the rename button, when file name is changed and not empty
+            if (editedValue !== files[index].name && editedValue !== "") {
+                this.setState({
+                    editedFileName: editedValue,
+                    enableButton: index,
+                });
+            }
+        }
+    };
+
+    // Function triggered when rename button is clicked
+    onFileRename = (index: number) => {
+        const {files, fileName, editedFileName} = this.state;
+        const currentFile = files[index];
+        // creating a file with newly provided name, since the file object is immutable
+        const blob = currentFile.slice(0, currentFile.size, currentFile.type);
+        const file = new File([blob], editedFileName, {type: currentFile.type});
+        // Replacing the new file with the old file
+        files[index] = file;
+        // Renaming in the fileName string
+        const fileNameArray = fileName.split(",");
+        fileNameArray[index] = file.name;
+        this.setState(
+            {
+                files,
+                fileName: fileNameArray.join(","),
+                editedFileName: "",
+                enableButton: ENABLE_BUTTON_DEFAULT,
+            },
+            () => this.props.onFileChange(this.state.files),
+        );
+    };
+
+    // Function triggered when remove button is clicked
+    onFileRemove = (index: any) => {
+        const files = this.state.files.slice();
```

Comment:
Please follow a consistent coding style. Extract all variables from props and state at beginning of the function.
this comment is applicable to all functions